server/services/transactionsTransformer_test.go

```diff
@@ -391,41 +458,84 @@ func TestTransactionsTransformer_InvalidTxToRosettaTx(t *testing.T) {
 	extension := newNetworkProviderExtension(networkProvider)
 	transformer := newTransactionsTransformer(networkProvider)
 
-	tx := &transaction.ApiTransactionResult{
-		Hash:             "aaaa",
-		Sender:           testscommon.TestAddressAlice,
-		Receiver:         testscommon.TestAddressBob,
-		Value:            "1234",
-		Type:             string(transaction.TxTypeInvalid),
-		InitiallyPaidFee: "50000000000000",
-	}
+	t.Run("when fee payer is sender", func(t *testing.T) {
+		tx := &transaction.ApiTransactionResult{
+			Hash:             "aaaa",
+			Sender:           testscommon.TestAddressAlice,
+			Receiver:         testscommon.TestAddressBob,
+			Value:            "1234",
+			Type:             string(transaction.TxTypeInvalid),
+			InitiallyPaidFee: "50000000000000",
+		}
 
-	expectedTx := &types.Transaction{
-		TransactionIdentifier: hashToTransactionIdentifier("aaaa"),
-		Operations: []*types.Operation{
-			{
-				Status:  &opStatusFailure,
-				Type:    opTransfer,
-				Account: addressToAccountIdentifier(testscommon.TestAddressAlice),
-				Amount:  extension.valueToNativeAmount("-1234"),
-			},
-			{
-				Status:  &opStatusFailure,
-				Type:    opTransfer,
-				Account: addressToAccountIdentifier(testscommon.TestAddressBob),
-				Amount:  extension.valueToNativeAmount("1234"),
+		expectedTx := &types.Transaction{
+			TransactionIdentifier: hashToTransactionIdentifier("aaaa"),
+			Operations: []*types.Operation{
+				{
+					Status:  &opStatusFailure,
+					Type:    opTransfer,
+					Account: addressToAccountIdentifier(testscommon.TestAddressAlice),
+					Amount:  extension.valueToNativeAmount("-1234"),
+				},
+				{
+					Status:  &opStatusFailure,
+					Type:    opTransfer,
+					Account: addressToAccountIdentifier(testscommon.TestAddressBob),
+					Amount:  extension.valueToNativeAmount("1234"),
+				},
+				{
+					Type:    opFeeOfInvalidTx,
+					Account: addressToAccountIdentifier(testscommon.TestAddressAlice),
+					Amount:  extension.valueToNativeAmount("-50000000000000"),
+				},
 			},
-			{
-				Type:    opFeeOfInvalidTx,
-				Account: addressToAccountIdentifier(testscommon.TestAddressAlice),
-				Amount:  extension.valueToNativeAmount("-50000000000000"),
+			Metadata: extractTransactionMetadata(tx),
+		}
+
+		rosettaTx := transformer.invalidTxToRosettaTx(tx)
+		require.Equal(t, expectedTx, rosettaTx)
+	})
+
+	t.Run("when fee payer is relayer", func(t *testing.T) {
```

Comment:
can you add a test where the relayer is also one of the sender or receiver?

Reply:
Added.